engine/commands/engine_init_cmd.cc

```diff
@@ -224,15 +197,24 @@ bool EngineInitCmd::Exec() const {
 
             download_service.AddDownloadTask(
                 downloadCudaToolkitTask,
-                [](const std::string& absolute_path, bool unused) {
+                [this](const std::string& absolute_path, bool unused) {
                   LOG_DEBUG << "Downloaded cuda path: " << absolute_path;
                   // try to unzip the downloaded file
                   std::filesystem::path downloaded_path{absolute_path};
-
-                  archive_utils::ExtractArchive(
-                      absolute_path,
-                      downloaded_path.parent_path().parent_path().string());
-
+                  // TODO(any) This is a temporary fix. The issue will be fixed when we has CIs
+                  // to pack CUDA dependecies into engine release
+                  auto get_engine_path = [](std::string_view e) {
+                    if (e == "cortex.llamacpp") {
+                      return cortex_utils::kLlamaLibPath;
+                    } else if (e == "cortex.tensorrt-llm") {
+                      return cortex_utils::kTensorrtLlmPath;
+                    }
+                    return cortex_utils::kLlamaLibPath;
```

Comment:
Lines 208 and 212 - is it extraneous to have 208?

Reply:
> Lines 208 and 212 - is it extraneous to have 208?

We fallback to default, will change the function to make it clearer.